Sources/Classes/DataLoading/GeneralDataLoading/GeneralDataLoadingViewModel.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  Copyright (c) 2018 Touch Instinct
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the Software), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import RxSwift
+import RxCocoa
+
+open class GeneralDataLoadingViewModel<ResultType>: BaseViewModel {
+
+    public typealias LoadingModel = GeneralDataLoadingModel<ResultType>
+    public typealias DataSourceType = Single<ResultType>
+    public typealias LoadingState = GeneralDataLoadingState<DataSourceType>
+
+    private let loadingModel: LoadingModel
+
+    private let loadingStateVariable = Variable<LoadingState>(.initial)
+
+    private let disposeBag = DisposeBag()
```

Comment:
Public

Reply:
And remove `DisposeBagHolder`

Reply:
1] Did you mean to use view model dispose bag, instead of view controller dispose bag?
2] It's pretty usable for hud operations with Observable/Single (Farmy, Jet).